torch/csrc/api/include/torch/data/samplers/base.h

```diff
@@ -41,6 +41,40 @@ class Sampler {
   /// Deserializes the `Sampler` from the `archive`.
   TORCH_API virtual void load(serialize::InputArchive& archive) = 0;
 };
+
+/// Wraps a provided sampler to make it thread safe.
+template <typename OriginalSampler>
+class ThreadSafeSampler
```

Comment:
Don't have a strong opinion. `LockedSampler` is more precise, so seems slightly better.

Reply:
I personally feel ThreadSafeSampler is better as the name clearly states when to use it: if it is single thread, the name indicates to not bother with this; if it is an multi-thread scenario, this is the sampler the developer should be using. The locking is an unavoidable cost in the latter case. 

On the other hand, LockedSampler doesn't have such information on naming and requires developers to look into its implementation to decide whether to use or not.

Reply:
> The locking is an unavoidable cost in the latter case.

This is clearly a false statement. I can name a few different approaches to mutual exclusion and concurrency like atomics or transactional memory, which all achieve the same purpose -- thread safety. That's why `LockedSampler` is more precise.

I'm also not very afraid of having people confuse it with anything else, since what other kinds of locks could one apply to samplers?

Reply:
Some form of locking is used to make the sampler thread-safe. I don't have a strong preference, but locked sampler just make me wonder what it does. To make things more interesting, we could bring Synchronized or Concurrent into the mix too. In any case, let's call it LockedSampler :)